lib/utils/cloud.ts

```diff
@@ -284,3 +283,46 @@ export async function getFormattedOsVersions(
 	}
 	return versions;
 }
+
+/**
+ * Duplicate of the balena SDK getMaxSatisfyingVersion() function, modified
+ * to use 'balena-semver' instead of 'semver'.
+ *
+ * TODO: replace this function with balena SDK v16. Don't forget to test the new
+ * SDK with ESR versions as well, because e.g. '2021.01.01.prod' is also an
+ * invalid semver that the SDK needs to support.
+ *
+ * Background:
+ * Originally, if the user provided a balenaOS version as a command line option,
+ * that version would just be used as is. The CLI would not query the list of
+ * available OS versions to match against. This was OK until balenaOS v2.88.4
+ * that removed the '+rev' revision metadata component (semver), which caused
+ * balena SDK v15 to fail in the getMaxSatisfyingVersion() method because e.g.
+ * '2.88.4.prod' is an invalid semver. This could not be fixed in SDK v15 because
+ * SDK v16 was already released, and the CLI could not be updated to use SDK v16
+ * because SDK v16 dropped support for Node.js v10 that the CLI still needed
+ * while we were preparing the CLI's new major version 13. So we came up with
+ * this funciton as temporary workaround until we could update the CLI to use
+ * SDK v16.
+ *
+ * @param deviceType E.g. 'raspberrypi4-64'
+ * @param version E.g. '2.88.4'
+ * @param esr True if `version` is a balenaOS ESR version
+ * @returns E.g. '2.88.4.prod' or '2.88.4.dev' or 'v2.87.16+rev1.prod'
+ */
+async function getMaxSatisfyingVersion(
```

Comment:
@thgreasi, FYI, this is what I've come up with so far. Still draft, pending proper testing.